### PR TITLE
[client] Consolidate transfers

### PR DESCRIPTION
**Pull request by rhlsthrm**
## The Problem
#1086 

## Checklist:

- [ ] I am making this PR against staging, not master
- [ ] My code follows the code style of this project.
- [ ] I have described any backwards-incompatibility implications above.
- [ ] I have highlighted which parts of the code should be reviewed most carefully.
- [ ] If my change requires a change to the documentation, I have updated it accordingly.

### CODE REVIEW

File: modules/types/src/transfers.ts [hunk 1]
```diff
@@ -4,18 +4,40 @@ import {
...
-export const GenericConditionalTransferAppName = "GenericConditionalTransferApp";
+export type TransferType = "RequireOnline" | "AllowOffline";
+export const getTransferTypeFromAppName = (
+  name: SupportedApplications,
+): TransferType | undefined => {
+  switch (name) {
+    case SupportedApplications.DepositApp:
+    case SupportedApplications.SimpleTwoPartySwapApp:
+    case SupportedApplications.WithdrawApp: {
+      return undefined;
```
Comment (bohendo) [inline modules/types/src/transfers.ts:22]: It's not obvious what an undefined return value means here. If it doesn't apply, we could explicitly return a "NotApplicable" `TransferType`.
Additionally, it seems like we use this fn to tag just a few apps where it *is* relevant, therefore we should default to `undefined` for all others.

Defining exceptions out of existence is recommended by A Philosophy of Software design. The highlights I made while reading this book are available here: https://gitlab.com/bohendo/my-clippings/-/blob/master/highlights/a-philosophy-of-software-design.md
> loc 1827 - This chapter focused on exceptions, which are one of the most significant sources of special-case code, and discussed how to reduce the number of places where exceptions must be handled. The best way to do this is by redefining semantics to eliminate error conditions. For exceptions that can’t be defined away, you should look for opportunities to mask them at a low level, so their impact is limited, or aggregate several special-case handlers into a single more generic handler.

File: modules/client/src/controllers/ConditionalTransferController.ts [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+  PublicResults,
+  GenericConditionalTransferAppState,
+} from "@connext/types";
+import { toBN, stringify } from "@connext/utils";
+import { HashZero, Zero } from "ethers/constants";
+
+import { AbstractController } from "./AbstractController";
+import { soliditySha256 } from "ethers/utils";
+
+export class ConditionalTransferController extends AbstractController {
```
Comment (bohendo) [inline modules/client/src/controllers/ConditionalTransferController.ts:17]: Calling this a "ConditionalTransfer" Controller suggests that it's distinct from other types of transfer controllers. But, now that we've consolidated all transfers into one controller, the "Conditional" prefix isn't useful for distinguishing this type of transfer logic from others. I'd recommend calling this module "CreateTransferController" & it's sibling "ResolveTransferController" instead.

File: modules/client/src/connext.ts [hunk 3]
```diff
@@ -343,42 +321,19 @@ export class ConnextClient implements IConnextClient {
   public resolveCondition = async (
```
Comment (bohendo) [inline modules/client/src/connext.ts:321]: While we're refactoring I'd like to clean up some of these names. It's not obvious what `resolveCondition` does just from it's method name (what kind of condition?). Event the `transfer` method suggests an atomic operation where, in reality, there are two distinct steps. IMO two methods `createTransfer` and `resolveTransfer` are much more obvious. What do you think?

(If we do rename these methods, we should prob keep the old ones around too for backwards-compatibility but mark them as depreciated & flag for removal during our next breaking release)

File: modules/client/src/controllers/ConditionalTransferController.ts (hunk 2 above)
Comment (ArjunBhuptani) [reply modules/client/src/controllers/ConditionalTransferController.ts:17]: I agree with this generally but perhaps we should still have an atomic `transfer` method? We'd still need at least one type of transfer to be the "easy"/default transfer that requires minimal overhead.

File: modules/client/src/connext.ts (hunk 3 above)
Comment (rhlsthrm) [reply modules/client/src/connext.ts:322]: I held off on the interface breaking changes. I renamed the methods internally. We can do a separate cleanup around the interface later.

**Pull request merged**